### PR TITLE
Teleport opt-in adoption survey

### PR DESCRIPTION
**Pull request by benarent**
This draft PR provides instructions for users to opt-in submit anonymous usage stats. This is based off the internal RFD https://github.com/gravitational/peopleware/pull/84 .

This draft PR is looking for
- [x] Feedback from the Bash Gods. Currently tested locally on my Mac
- [x] UX feedback

This PR is waiting on migrating the API Gateway account to our official account. The backend is currently  `API  Gateway -> Lamba -> Aurora Serverless Postgres`.

### Example UX
```
➜  sudo ./install
Teleport binaries have been copied to /usr/local/bin

Thanks for installing Teleport. Is it OK if we collect some information about your install?

# Updated with your use case and optional email address if you would like to receive link to get swag
$ curl -X POST https://usage.teleport.dev -F OS=darwin -F use-case="secure databases"  -F email=alicea@example.com

Otherwise, ignore!
```

### CLI Feedback post survey submission. 
```
➜ curl -X POST https://usage.teleport.dev  -F OS=darwin -F use-case="secure databases" -F email=alice@example.com
Thank you for submitting Teleport usage data.
We will send you a meaningful newsletter and reach out to send some swag.
```

### CODE REVIEW

File: build.assets/install [hunk 1]
```diff
@@ -15,5 +15,31 @@ cd $(dirname $0)
...
+set -- $teleportVersion; version=$2$3
+
 echo "Teleport binaries have been copied to $BINDIR"
-echo "To configure the systemd service for Teleport take a look at examples/systemd/README.md"
+echo ""
+echo "Thanks for installing Teleport. Is it OK if we collect some metrics about your install?"
+echo ""
+echo "If you don't mind please run:"
+echo "# Update email if you would like to receive link to get swag"
+echo "$ curl -X POST https://9sjnq1g1j9.execute-api.us-west-2.amazonaws.com/Prod/usage -F OS=${machine} -F version=${version} -F email=alice@example.com"
```
Comment (benarent) [inline build.assets/install:43]: Note: This is a testing URL. We'll move to a shorter URL e.g. https://usage.teleport.dev

Comment (klizhentas) [reply build.assets/install:43]: Let's improve error messages:

```
curl -X POST https://9sjnq1g1j9.execute-api.us-west-2.amazonaws.com/Prod/usage
{"message": "Internal server error"}
```

Can you send a code in the same PR?

Comment (klizhentas) [reply build.assets/install:43]: Let's explain how we will use the data on submission:

```
curl -X POST https://9sjnq1g1j9.execute-api.us-west-2.amazonaws.com/Prod/usage -F email=alice@example.com
Thank you for submitting Teleport usage data. We will send you a meaningful newsletter and reach out to send some swag.
```

Comment (benarent) [reply build.assets/install:43]: I've fixed the error message if a user doesn't submit anything. I'll look at improving the message before submitting the survey.

**Comment by TravisGary** [comment]
@benarent Is there a list of exactly what data will be included in the opt in collection?  

Is this similar to the resource usage that cloud collects for billing purposes?  I.e. how many servers, users, databases, etc are connected?

**Comment by benarent** [comment]
> Is this similar to the resource usage that cloud collects for billing purposes? I.e. how many servers, users, databases, etc are connected?

@TravisGary  The survey will only collect version, operating system and email address. I would suggest we create a dedicate table for this. I've a serverless GoLang demo. I'll open up a ticket for how to go about getting it deployed in our infra.

**Comment by benarent** [comment]
This PR is now ready for review, `https://usage.teleport.dev` has been setup in our prod AWS account and has a dedicated Aurora Serverless Postgres Database. 

The current API gateway has basic validation and requires OS and Version to be present. It currently provides limited rate limiting, and if we find abuse on the endpoint we can add an [AWS WAF](https://aws.amazon.com/blogs/compute/amazon-api-gateway-adds-support-for-aws-waf/) to this endpoint.

**Review by russjones — COMMENTED** [review]
@klizhentas Can you review?

**Comment by benarent** [comment]
A few updates on the nits.

1. I've added a max fixed length for email address.  This should be deployed ( 30min after this comment ) 
2. The API is responding with `{"message":"Too Many Requests"}%`  when using `zsh`, this is the same [issue as this](https://stackoverflow.com/questions/29497038/why-does-a-curl-request-return-a-percent-sign-with-every-request-in-zsh) I tried to modify API Gateway to add a new line but was unable to come up with something. I could add this to the Go Lambda but it's cheaper and easier to stop at the API Gateway.

**Review by klizhentas — CHANGES_REQUESTED** [review]
Sorry for the late change. I've been thinking about this and I think version is much less relevant than use case. Can we replace version field with use-case field so we can learn more about use-cases of folks using teleport?

**Comment by benarent** [comment]
Sure, it's easy to change. I'm a little concerned that the inline form is a little clunky. Maybe we send these as variables or make the survey interactive?  

If not, how's the example below? 

```
➜  sudo ./install
Teleport binaries have been copied to /usr/local/bin

Thanks for installing Teleport. Is it OK if we collect some information about your install?

# Updated with your use case and optional email address if you would like to receive link to get swag
$ curl -X POST https://usage.teleport.dev -F os=darwin -F use-case="secure databases"  -F email=alicea@example.com

Otherwise, ignore!
```

**Comment by klizhentas** [comment]
@benarent I like that it's hacky, part of the appeal :) invites you to submit the form a couple of times to test it, so let's go with this format

**Comment by benarent** [comment]
The Survey has been updated to support use-case.

```
➜  ~ curl -X POST https://usage.teleport.dev  -F OS=darwin -F use-case="secure databases" -F email=alice@example.com
Thank you for submitting you Teleport use case.
We will send you a meaningful newsletter and reach out to send some swag.
```
```
# If a very long reason is supplied
➜  ~ curl -X POST https://usage.teleport.dev  -F OS=darwin -F use-case="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum" -F email=alice@example.com
That's a very long reason why. Please let us know more in our community https://github.com/gravitational/teleport/discussions
```

**Comment by benarent** [comment]
@klizhentas Can you PTAL again. It needs re-approving since you request changes.

**Review by klizhentas — APPROVED** [review]

**Pull request merged**